api/src/repositories/survey-block-repository.ts

```diff
@@ -0,0 +1,147 @@
+import SQL from 'sql-template-strings';
+import { z } from 'zod';
+import { ApiExecuteSQLError } from '../errors/api-error';
+import { BaseRepository } from './base-repository';
+
+export const SurveyBlock = z.object({
+  survey_block_id: z.number().nullable(),
+  survey_id: z.number(),
+  name: z.string(),
+  description: z.string()
+});
+
+export type SurveyBlock = z.infer<typeof SurveyBlock>;
+
+// This describes the a row in the database for Survey Block
+export const SurveyBlockRecord = z.object({
```

Comment:
Maybe could use `z.extend` here?

Reply:
Comment is old now but I think it's better in this case to leave these as 2 discrete type definitions instead of extending one of them, keeps it a little easier to read in my opinion.